tests/suite/system_suite_test.go

```diff
@@ -96,8 +96,10 @@ type setupConfig struct {
 	gwAPIVersion string
 	deploy       bool
 	nfr          bool
+	infoLogLevel bool
 }
 
+//nolint:gocyclo
```

Comment:
Would get this error/warning if I don't skip gocyclo:

`cyclomatic complexity 16 of func `setup` is high (> 15) (gocyclo)`

Are we fine adding this skip?

Reply:
If there is a way to break up the function into smaller functions, that will fix this issue.